.eslintrc.yml

```
@@ -0,0 +1,92 @@
+extends:
+  - plugin:@typescript-eslint/eslint-recommended
+  - plugin:@typescript-eslint/recommended
+  - plugin:@typescript-eslint/recommended-requiring-type-checking
+  - plugin:import/errors
+  - plugin:import/warnings
+  - plugin:import/typescript
+  - plugin:promise/recommended
+  - airbnb-typescript
+  - plugin:jest/recommended
+  - plugin:jest/style
+  - plugin:prettier/recommended
+plugins:
+  - react-hooks
+  - unicorn
+parser: "@typescript-eslint/parser"
+parserOptions:
+  ecmaVersion: 6
+  sourceType: module
+  ecmaFeatures:
+    jsx: true
+    modules: true
+  project: "./tsconfig.json"
+env:
+  node: true
+  es6: true
+  browser: true
+rules:
+  # NOTE: any rule set to warning (warn or 1)
+  # eslint will NOT  output any warning when running `npm test` or `npm run eslint`
+  # warnings will still appear on the editor 
+  no-nested-ternary: 0
+  curly: error
+  "@typescript-eslint/no-unused-vars": error
+  "@typescript-eslint/no-floating-promises": error
+  react/jsx-wrap-multilines:
+    - error
+    - declaration: false
+      assignment: false
+  react/jsx-indent: 0
+  react/jsx-props-no-spreading: 0
+  react/prop-types: 0
+  react-hooks/rules-of-hooks: error
+  react/no-array-index-key: error 
+  react/jsx-indent-props: 0
+  react/no-unescaped-entities: 0
+  import/no-unresolved: error #re-add no-unresolved
+  unicorn/custom-error-definition: error
+  import/no-extraneous-dependencies:
+    - error
+    - devDependencies:
+      - '**/*.spec.{ts,tsx}'
+      - '**/*.stories.{ts,tsx}'
+      - '.storybook/*.{js,jsx,ts,tsx}'
+      - 'jest.*.js'
+      - types/**/*.d.ts
+      - scripts/**/*.{ts,tsx,js,jsx}
+  react/jsx-one-expression-per-line: 0
+  react/jsx-key: error
+  import/prefer-default-export: 0
+  quotes:  
+    - error
+    - single # has to match prettier quotes
+    - avoidEscape: true # has to always be true to not clash with preetier
```

I'm curious why we need formatting rules like this for eslint. I'd imagine they'd fall under the purview of prettier.

```suggestion
    - avoidEscape: true # has to always be true to not clash with prettier
```